Unable to add credentials to existing Docker Hub endpoint

Hello.

Not sure if this is by design but wanted to report anyway.

After creating a Docker Hub registry endpoint with NO application ID and secret, I wasn't able to edit the endpoint and add the credentials to it so that I would be able to access private images. The web UI would just say "Internal Server Error". This is the relevant error message from core.log (I triple checked that the credentials were correct):

```
Aug  2 14:48:25 172.26.0.1 core[2910616]: 2021-08-02T12:48:25Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: login to dockerhub error: login to dockerhub error: {\"detail\":\"Incorrect authentication credentials\"}\n\n"}]}
```

I had to delete the endpoint and create a new one with credentials, which worked.

If this is by design a clearer error message would help here.

**Versions:**

- harbor version: 2.3.0
- docker engine version: 20.10.6
- docker-compose version: 1.29.2

--- a/src/portal/src/app/base/left-side-nav/registries/create-edit-endpoint/create-edit-endpoint.component.ts
+++ b/src/portal/src/app/base/left-side-nav/registries/create-edit-endpoint/create-edit-endpoint.component.ts
@@ -423,7 +423,7 @@ export class CreateEditEndpointComponent
     if (!this.target || !this.initVal) {
       return changes;
     }
-    for (let prop of Object.keys(this.target)) {
+    for (let prop of Object.keys(Object.assign({}, this.target, this.initVal))) {
       let field: any = this.initVal[prop];
       if (typeof field !== "object") {
         if (!compareValue(field, this.target[prop])) {
@@ -439,7 +439,7 @@ export class CreateEditEndpointComponent
           }
         }
       } else {
-        for (let pro of Object.keys(field)) {
+        for (let pro of Object.keys(Object.assign({}, field, this.target[prop]))) {
           if (!compareValue(field[pro], this.target[prop][pro])) {
             changes[pro] = this.target[prop][pro];
             // Number

--- a/src/portal/src/app/shared/units/utils.ts
+++ b/src/portal/src/app/shared/units/utils.ts
@@ -567,7 +567,7 @@ export const validateLimit = unitContrl => {
 };
 
 export function formatSize(tagSize: string): string {
-    const size: number = Number.parseInt(tagSize);
+    const size: number = Number.parseInt(tagSize, 10);
     if (Math.pow(1024, 1) <= size && size < Math.pow(1024, 2)) {
         return (size / Math.pow(1024, 1)).toFixed(2) + "KiB";
     } else if (Math.pow(1024, 2) <= size && size < Math.pow(1024, 3)) {

